docs/architecture/blazor-for-web-forms-developers/migration.md

```diff
@@ -8,8 +8,322 @@ ms.date: 09/11/2019
+`HttpContext.Current` no longer exist and must be accessed via the provider
+`IHttpContextAccessor.HttpContext`.
+
+Tasks are used throughout the stack and should be used all the way down.
+Although making a method return a `Task` is a viral and can affect all calling
+methods, any place where `Task.Wait()` or similar is called can adversely affect
+performance. This is a blocking call and will cause threads to be unuseable by
+the rest of the pipeline. Since the ThreadPool is used for this, it can cause
+performance degradation as threads become locked up and are unavailable for
+queued work.
+
+### Built in general abstractions 
+
+ASP.NET Core provides many general abstractions that allow for a clean
+separation of service from implementation. Services like logging, configuration,
+and dependency injection can be abstracted away via some common interfaces that
+have default implementations or can wrap well-known 3rd party libraries. For
+example, although the logging infrastructure by default outputs with a simple
+logging provider, it can easily be replaced with anything else, including
+log4net, NLog, and Serilog (among many others).
+
+### Centralized start up method
+
+In WebForms, there there a number of ways to configure the application from `web.config`, `Global.asax`, and Owin. In ASP.NET Core, this is simplified to a single startup class that handles two steps:
+
+- Configure services required in the application
+  
+  Dependency injection is built into ASP.NET Core from the bottom up so
+  everything can be considered a service that can be scoped how needed. This can
+  be per-instance, per-request, or an application singleton. This startup method
+  allows you to add services for your own application, as well as override
+  system ones with your own implementation.
+
+- Configure pipeline of middleware
+
+  Middleware can be thought of as handlers that take a request in and has the
+  ability to process it before or after sending it on to the next piece of
+  middleware. There are many built in middleware that can be added into the
+  pipeline, as well as easy ways to add your own implementations to do things
+  like inspect or add headers, exception handling, logging, and many more. This
+  will be discussed more in depth in the chapter on
+  [Middleware](./middleware.md).
+
+## Example of code changes
+
+We'll demonstrate some of these code changes by looking at the [eShopOnBlazor](https://github.com/dotnet-architecture/eShopOnBlazor) project. This project is a simple eShop that allows a user to perform simple CRUD operations on a catalog. We will look specifically at the Create page and the changes required to make that work.
+
+### Start up changes
+
+In Blazor, the startup process is the same as other ASP.NET Core frameworks.
+This is greatly simplified from traditional ASP.NET where everything that is
+included is in the start up class itself. Comparing `Global.asax.cs` vs
```

Comment:
You don't really need to compare `Global.asax` and `Startup` here as that should already have been covered in the "Startup" chapter. Maybe go ahead and move this content there?